### PR TITLE
fix(ui): UI bug fixes

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Review by simster7 — APPROVED** [review]

File: ui/src/app/event-flow/components/event-flow-details/event-flow-page.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const EventFlowPage = ({history, location, match}: RouteComponentProps<an
     // state for URL and query parameters
     const [namespace, setNamespace] = useState(match.params.namespace || '');
     const [showFlow, setShowFlow] = useState(queryParams.get('showFlow') === 'true');
-    const [showWorkflows, setShowWorkflows] = useState(queryParams.get('showWorkflows') === 'true');
```
Comment (alexec) [inline ui/src/app/event-flow/components/event-flow-details/event-flow-page.tsx:41]: default this low risk feature on

File: ui/src/app/shared/services/info-service.ts [hunk 2]
```diff
@@ -2,9 +2,15 @@ import {GetUserInfoResponse, Info, Version} from '../../../models';
 
 import requests from './requests';
 
+let info: Promise<Info>; // we cache this globally rather than in localStorage so it is request once per page refresh
```
Comment (alexec) [inline ui/src/app/shared/services/info-service.ts:5]: cache info in memory so we don't retrieve it repeatedly

File: ui/src/app/userinfo/components/cli-help.tsx [hunk 3]
```diff
@@ -29,36 +29,36 @@ argo list`;
...
+                            </>
+                        ) : (
+                            <>
+                                <i className='fa fa-copy' /> Copy to clipboard
+                            </>
+                        )}
+                    </button>
+                </div>
+            </Notice>
+            <textarea ref={hiddenText} style={{width: 0, height: 0, opacity: 0}} defaultValue={text} />
```
Comment (alexec) [inline ui/src/app/userinfo/components/cli-help.tsx:61]: remove a React warning

File: ui/src/app/widgets/widget-gallery.tsx [hunk 4]
```diff
@@ -47,7 +47,7 @@ export const WidgetGallery = ({namespace, name, label}: {namespace: string; name
                 </div>
                 <ul>
                     {parameters.map(p => (
-                        <li>
+                        <li key={p.name}>
```
Comment (alexec) [inline ui/src/app/widgets/widget-gallery.tsx:50]: remove a React warning

File: ui/src/app/workflows/components/workflow-creator.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ export const WorkflowCreator = ({namespace, onCreate}: {namespace: string; onCre
             .list(namespace)
             .then(setWorkflowTemplates)
             .catch(setError);
-    }, []);
+    }, [namespace]);
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-creator.tsx:29]: correct deps

File: ui/src/app/workflows/components/workflow-details/workflow-details.tsx [hunk 6]
```diff
@@ -195,7 +196,7 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
         );
         retryWatch.start(workflow.metadata.resourceVersion);
         return () => retryWatch.stop();
-    }, [workflow]);
+    }, [namespace, name]);
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-details/workflow-details.tsx:199]: this was a pretty nasty bug, every time we got a changed workflow, we re-established the watch, resulting in a hot loop

File: ui/src/app/workflows/components/workflow-details/workflow-details.tsx [hunk 7]
```diff
@@ -81,7 +81,8 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
                                         if (workflowOperation.title === 'DELETE') {
                                             navigation.goto(uiUrl(`workflows/${workflow.metadata.namespace}`));
                                         } else {
-                                            navigation.goto(uiUrl(`workflows/${wf.metadata.namespace}/${wf.metadata.name}`));
+                                            // navigation.goto does not seem to work here
+                                            document.location.href = uiUrl(`workflows/${wf.metadata.namespace}/${wf.metadata.name}`);
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-details/workflow-details.tsx:85]: not sure why, not a big enough issue to properly

**Pull request merged**